src/auth/digest/Config.cc

```diff
@@ -967,13 +967,11 @@ Auth::Digest::Config::decode(char const *proxy_auth, const HttpRequest *request,
             return rv;
         }
     } else {
-        /* cnonce and nc both require qop */
-        if (digest_request->cnonce || digest_request->nc[0] != '\0') {
-            debugs(29, 2, "missing qop!");
-            rv = authDigestLogUsername(username, digest_request, aRequestRealm);
-            safe_free(username);
-            return rv;
-        }
+        /* rfc7616 section 3.3 mandates qop to be present */
```

Comment:
Section 3.3 does not apply. This logic handles HTTP Request from the client.
As such the requirements to follow are in [RFC 7616 Section 3.4](https://www.rfc-editor.org/rfc/rfc7616#section-3.4).

```suggestion
        /* RFC7616 section 3.3, qop:
         * "MUST be used by all implementations"
         *
         * RFC7616 section 3.4, qop:
         *  "value MUST be one of the alternatives the server
         *   indicated it supports in the WWW-Authenticate header field"
         *
         * Squid sends qop=auth, reject buggy or outdated clients.
        */
```